learning/02_graphics/how_to_use_glm.markdown

```diff
@@ -0,0 +1,121 @@
+---
+.. title: The new GLM syntax 
+.. summary: How to use glm vectors instead the legacy ones
+.. type: howto
+---
+
+*Note!*: This page is useful only if you are working with the current master branch. You can skip this information if you are working with the version 0.9.3 that you have downloaded from this website.
+
+The next version of openFrameworks will replace the internal math library with [GLM](http://glm.g-truc.net). GLM is a solid C++ library used for all the math operations needed when doing vectors and matrices operations. The use of this library implies some change in the syntax used to declare vectors and to execute vector's operation. The legacy mode is still supported, but the new mode, enabled by default, uses the new glm syntax.
+
+If you are not interested using this library and you want to continue using the syntax you were used to, you can easily disable it using the OF_USE_LEGACY_MESH constant in ofConstants.h
+
+Instead, if you want to use GLM and prepare yourself for what will be the future syntax adopted by openFrameworks, these are the things that are changed:
+
+### Declaring the vectors
+
+When declaring vectors in the header file, you have to change
+
+```c++
+ofVec3f myVector;
+ofVec2f my2dVector;
+```
+
+To:
+
+```c++
+glm::vec3 myVector
+glm::vec2 my2dVector;
+```
+
+
+When the vectors come, for example, from a mesh, you don't have to worry about the type declaration, since c++ 11 you have the wonderful `auto` keywords, that tells to the compiler to figure out the type for you. For example, if you have code like:
+
+```c++
+vector<ofVec3f> normals = mesh.getNormals();
+```
+
+And you want to migrate it to the new mode, just use auto
+
+```c++
+auto normals = mesh.getNormals();
+```
+
+### Methods names
+
+Most methods have the exact same name but without camel case and others are slightly different:
```

Comment:
addressed in https://github.com/openframeworks/ofSite/pull/555/commits/4024021de5c843fb704ba6044b2af883738e9321